server.go

```diff
@@ -77,13 +77,25 @@ const defaultTCPReadTimeout = 10 * time.Minute
 
 const httpQuitEndpoint = "/quitquitquit"
 
-type SpanSinkTypes = map[string]func(
-	*Server, string, *logrus.Entry, Config, interface{},
-) (sinks.SpanSink, error)
+type SpanSinkTypes = map[string]struct {
+	// Creates a new span sink intsance.
+	Create func(
+		*Server, string, *logrus.Entry, Config, interface{},
+	) (sinks.SpanSink, error)
+	// Parses the config for the sink into a format that is validated and safe to
+	// log.
+	ParseConfig func(interface{}) (interface{}, error)
```

Comment:
Can we use a named interface here instead of just raw `interface{}`? Define `type SinkConfig interface{}` somewhere and then:

```suggestion
	ParseConfig func(interface{}) (SinkConfig, error)
```

Reply:
Oh nice, this is a good way to indicate where the value should be used. Thanks!